src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs

```diff
@@ -39,18 +40,94 @@ public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out TVal
                 return false;
             }
 
-            if (s_isUint64)
-            {
-                if (reader.TryGetUInt64(out ulong ulongValue))
-                {
-                    value = (TValue)Enum.ToObject(valueType, ulongValue);
-                    return true;
-                }
-            }
-            else if (reader.TryGetInt64(out long longValue))
+            // When utf8reader/writer will support all primitive types we should remove custom bound checks
+            // https://github.com/dotnet/corefx/issues/36125
+
+            switch (s_enumTypeCode)
             {
-                value = (TValue)Enum.ToObject(valueType, longValue);
-                return true;
+                case TypeCode.Empty:
```

Comment:
I do not think the unhandled cases need to be enumerated here.

Reply:
ok